gustaf/io/mfem.py

```diff
@@ -1,11 +1,206 @@
+        ]
+        vdim = int(lines[indices[-1] + 2])
+        # Extract values
+        elements = extract_values(
+                fname, indices[1] + 2, n_elements, total_lines,
+                settings.INT_DTYPE
+        )
+        if elements.shape[0] != n_elements:
+            raise ValueError("Number of elements do not match.")
+        boundary = extract_values(
+                fname, indices[2] + 2, n_boundaries, total_lines,
+                settings.INT_DTYPE
+        )
+        if boundary.shape[0] != n_boundaries:
+            raise ValueError("Number of boundaries do not match.")
+        vertices = extract_values(
+                fname, indices[3] + 3, n_vertices, total_lines,
+                settings.FLOAT_DTYPE
+        )
+        if vertices.shape != (n_vertices, vdim):
+            raise ValueError("Number of vertices do not match.")
+
+        connectivity = elements[:, 2:]
+        if dimension == 2:
+            mesh = Faces(vertices=vertices, faces=connectivity)
+        elif dimension == 3:
+            mesh = Volumes(vertices=vertices, volumes=connectivity)
+
+        return mesh
+
+
+def export_mesh(mesh, fname):
+    """
+    Export mesh in MFEM format.
+    Supports 2D triangle and quadrilateral meshes.
+    Does not support different element attributes or
+    difference in vertex dimension and mesh dimension
+
+    Parameters
+    ------------
+    mesh: Faces
+    fname: str
+
+    Returns
+    ------------
+    None
+    """
+    # Basic infos
+    nvertices, dim = mesh.vertices.shape
+
+    def format_array(array):
+        """
+        Format NumPy array into string.
+        Each entry in a row is separated by a blank space
+        and every row is separated by a new line.
+
+        Parameters
+        ------------
+        array: NumPy array
+
+        Returns
+        ------------
+        str
+        """
+        row_strings = [" ".join(map(str, row)) for row in array]
+        return "\n".join(row_strings)
+
+    # Export 2D mesh
+    if dim == 2:
+        # Elements
+        element_attribute = 1  # Other numbers not yet supported
+        elements = mesh.elements()
+        n_elements, n_element_vertices = elements.shape
+        if n_element_vertices == 3:
+            geometry_type = geometry_types["TRIANGLE"]
+        elif n_element_vertices == 4:
+            geometry_type = geometry_types["SQUARE"]
+        else:
+            raise NotImplementedError(
+                    f"Sorry, we cannot export mesh with elements \
+consisting of {n_element_vertices} vertices."
+            )
+        e = np.ones((n_elements, 1), dtype=settings.INT_DTYPE)
+        elements_array = np.hstack(
+                (element_attribute * e, geometry_type * e, elements)
+        )
+        elements_array_string = format_array(elements_array)
+        elements_string = f"elements\n{n_elements}\n"
+        elements_string += f"{elements_array_string}\n\n"
+
+        # Boundary
+        edges = mesh.get_edges()
```

Comment:
did you rebase? `get_*` isn't supported anymore.
```suggestion
        edges = mesh.edges()
```